packages/cisco_ftd/data_stream/log/elasticsearch/ingest_pipeline/default.yml

```
@@ -1921,6 +1919,52 @@ processors:
             !ctx._temp_.internal_zones.contains(ctx.observer.ingress.zone)
           )
         )
+  #
+  # Network Directionality when zones are not configured. 
+  #
+  # Requires Integration option "consider_private_internal" (defaults to True)
+  #
+  - script:
+      description: Calculate network.direction if zones are not configured
+      lang: painless
+      if: "ctx?._temp_?.consider_private_internal == true &&
+           ctx?.source?.ip != null &&
+           ctx?.destination?.ip != null &&
+          (ctx?._temp_?.external_zones == null ||
+           ctx?._temp_?.internal_zones == null)"
```

Rather than having non-orthogonal behaviour like this where the configurations interact with each other, I'd suggest this.
```suggestion
  #
  # Network Directionality when zones do not resolve direction. 
  #
  # Requires Integration option "consider_private_internal" (defaults to True)
  #
  - script:
      description: Calculate network.direction if zones have not resolved direction.
      lang: painless
      if: 'ctx._temp_?.consider_private_internal == true && 
           ctx.source?.ip != null && ctx.destination?.ip != null &&
          (ctx.network?.direction == null || ["", "unknown"].contains(ctx.network.direction))'
```
This means that if both are configured private ranges can be used in conjunction with the known zones, falling back to the private ranges if the known zones fail to resolve direction.

This behaviour (or the behaviour in the current proposal) needs to be described in the user-facing documentation.

---

Unfortunately, your suggested change will still end up with incorrect network direction. 

The Grok parsing of these messages already [incorrectly] assigned `network.direction`, so the value will never be null or "". After that, if Zones are not configured, the zone-based logic will not run, so it will also never be "unknown". With that conditional you added, the script will never get run.

---

We can remove the network.direction setting in the grok.

---

I have only identified this `network.direction` issue in the two Message ID's noted in this PR, as they seem to be the most common Message ID's in our environment, but given the admission from Cisco, the issue would have to be present for all Message ID's that include a direction string in the message. Are you suggesting I remove the `network.direction` setting in **_all_** Grok **_and_** Dissect patterns? That would be the only way your suggested change would work. I also don't know what "non-orthogonal" means, hahaha.

---

Yes, that's my suggestion.

Orthogonality of features is when changing one does not affect the other. This is generally a fairly helpful thing to avoid surprises. When I'm driving my car, I don't want the airconditioner control to interact with the brakes for example. In this particular case making a change in one setting give surprising behaviour from another.